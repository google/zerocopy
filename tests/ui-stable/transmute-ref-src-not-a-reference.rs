../ui-nightly/transmute-ref-src-not-a-reference.rs