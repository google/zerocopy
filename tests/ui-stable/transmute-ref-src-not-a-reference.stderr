error[E0308]: mismatched types
  --> tests/ui-stable/transmute-ref-src-not-a-reference.rs:17:49
   |
17 | const SRC_NOT_A_REFERENCE: &u8 = transmute_ref!(0usize);
   |                                  ---------------^^^^^^-
   |                                  |              |
   |                                  |              expected `&_`, found `usize`
   |                                  expected due to this
   |
   = note: expected reference `&_`
                   found type `usize`
help: consider borrowing here
   |
17 | const SRC_NOT_A_REFERENCE: &u8 = transmute_ref!(&0usize);
   |                                                 +

warning: never type fallback affects this call to an `unsafe` function
  --> tests/ui-stable/transmute-ref-src-not-a-reference.rs:17:34
   |
17 | const SRC_NOT_A_REFERENCE: &u8 = transmute_ref!(0usize);
   |                                  ^^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this will change its meaning in a future release!
   = note: for more information, see issue #123748 <https://github.com/rust-lang/rust/issues/123748>
   = help: specify the type explicitly
   = note: `#[warn(never_type_fallback_flowing_into_unsafe)]` on by default
   = note: this warning originates in the macro `$crate::assert_size_eq` which comes from the expansion of the macro `transmute_ref` (in Nightly builds, run with -Z macro-backtrace for more info)

warning: never type fallback affects this call to an `unsafe` function
  --> tests/ui-stable/transmute-ref-src-not-a-reference.rs:17:34
   |
17 | const SRC_NOT_A_REFERENCE: &u8 = transmute_ref!(0usize);
   |                                  ^^^^^^^^^^^^^^^^^^^^^^
   |
   = warning: this will change its meaning in a future release!
   = note: for more information, see issue #123748 <https://github.com/rust-lang/rust/issues/123748>
   = help: specify the type explicitly
   = note: this warning originates in the macro `transmute_ref` (in Nightly builds, run with -Z macro-backtrace for more info)
